Stack/Opc.Ua.Core/Stack/Server/EndpointBase.cs

```diff
@@ -822,6 +823,96 @@ public static IServiceRequest GetRequest(IAsyncResult ar)
             #endregion
 
             #region Private Members
+
+            /// <summary>
+            /// Activity Source Name.
+            /// </summary>
+            public static readonly string ActivitySourceName = "Opc.Ua.Server-ActivitySource";
+
+            /// <summary>
+            /// Activity Source static instance.
+            /// </summary>
+            public static ActivitySource ActivitySource => s_activitySource.Value;
+            private static readonly Lazy<ActivitySource> s_activitySource = new Lazy<ActivitySource>(() => new ActivitySource(ActivitySourceName, "1.0.0"));
+
+            /// <summary>
+            /// Holds the tracing context for propagation across system boundaries.
+            /// </summary>
+            public class TraceContext
+            {
+                /// <summary>
+                /// Gets the core trace identifiers like TraceId and SpanId.
+                /// </summary>
+                public ActivityContext ActivityContext { get; }
+
+                /// <summary>
+                /// Gets the user-defined data associated with the trace.
+                /// </summary>
+                public Dictionary<string, string> Baggage { get; }
+
+                /// <summary>
+                /// Constructs a new TraceContext.
+                /// </summary>
+                /// <param name="activityContext">Core trace identifiers.</param>
+                /// <param name="baggage">User-defined data for the trace.</param>
+                public TraceContext(ActivityContext activityContext, Dictionary<string, string> baggage)
+                {
+                    ActivityContext = activityContext;
+                    Baggage = baggage;
+                }
+            }
+
+            /// <summary>
+            /// Extracts the trace and baggage details from the given dictionary
+            /// </summary>
+            public static TraceContext ExtractTraceContextFromParameters(AdditionalParametersType parameters)
```

Comment:
try to make this function with as little overhead as possible. I don't see where baggage is needed?

Reply:
improved the function